### PR TITLE
ref chameleon-system/chameleon-system#2: load media manager url from …

### PR DESCRIPTION
**Pull request by pixeljunkie**
…media manager url generator in image field mapper

| Q             | A
| ------------- | ---
| Branch?       | 6.2.x
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#2 
| License       | MIT

### CODE REVIEW

File: src/CoreBundle/private/library/classes/mappers/TCMSMediaFieldMapper.class.php [hunk 1]
```diff
@@ -9,8 +9,33 @@
...
+     */
+    private $mediaManagerUrlGenerator;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param MediaManagerUrlGeneratorInterface $mediaManagerUrlGenerator
```
[the remark this answers is not in the document]
Comment (pixeljunkie) [reply src/CoreBundle/private/library/classes/mappers/TCMSMediaFieldMapper.class.php]: I reverted the whole change in this class. See later comment why.

File: src/CoreBundle/private/library/classes/mappers/TCMSMediaFieldMapper.class.php [hunk 2]
```diff
@@ -27,16 +52,33 @@ public function GetRequirements(IMapperRequirementsRestricted $oRequirements)
...
+        $mediaManagerButtonHtml = $this->getMediaManagerButtonHtml();
+        $oVisitor->SetMappedValue('sHtmlManageMediaButton', $mediaManagerButtonHtml);
         $oVisitor->SetMappedValue('sFieldName', $sFieldName);
         $oVisitor->SetMappedValue('sRecordId', $sRecordId);
         $oVisitor->SetMappedValue('sTableId', $sTableId);
         $oVisitor->SetMappedValue('sHtmlHiddenFields', $sHtmlHiddenFields);
     }
+
+    /**
+     * @return string
```
[the remark this answers is not in the document]
Comment (pixeljunkie) [reply src/CoreBundle/private/library/classes/mappers/TCMSMediaFieldMapper.class.php]: I reverted the whole change in this class. See later comment why.

File: src/CoreBundle/Resources/config/mappers.xml [hunk 3]
```diff
@@ -22,6 +22,8 @@
         </service>
 
         <service id="chameleon_system_core.mapper.media_field" class="TCMSMediaFieldMapper">
+            <argument type="service" id="chameleon_system_core.media_manager.url_generator" />
```
Comment (umulmrum) [inline src/CoreBundle/Resources/config/mappers.xml]: Those lines are no longer required due to the new decorator.

Comment (pixeljunkie) [reply src/CoreBundle/Resources/config/mappers.xml]: reverted the change.

**Comment by umulmrum** [comment]
Thanks @pixeljunkie

**Pull request merged**